### PR TITLE
thrift: add version 0.16.0 and update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **thrift/0.16.0**

- update boost/1.79.0
- enable with_qt (renamed as with_qt5 because cmake option is `WITH_QT5`
- some cmake options(WITH_STATIC_LIB, WITH_SHARED_LIB) are deprecated since 0.12.0. So they are removed.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
> fatal error C1902: Program database manager mismatch; please check your installation 

oh...

File: recipes/thrift/all/conanfile.py [hunk 1]
```diff
@@ -161,7 +146,7 @@ def package(self):
             targets.update({"thriftz::thriftz": "thrift::thriftz"})
```
Comment (madebr) [inline recipes/thrift/all/conanfile.py:146]: Not related tho this pr, but should we add `lib/cmake` to `self.cpp_info.components["libthrift"].builddirs`?

**Review by SSE4 — APPROVED** [review]

File: recipes/thrift/all/conanfile.py [hunk 2]
```diff
@@ -200,11 +185,14 @@ def package_info(self):
...
         if self.settings.os == "Windows":
             self.cpp_info.components["libthrift"].defines.append("NOMINMAX")
+            if tools.Version(self.version) >= "0.15.0":
+                self.cpp_info.components["libthrift"].system_libs.append("shlwapi")
         elif self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.components["libthrift"].system_libs.extend(["m", "pthread"])
         self.cpp_info.components["libthrift"].requires.append("boost::headers")
         if self.options.with_openssl:
             self.cpp_info.components["libthrift"].requires.append("openssl::openssl")
+        self.cpp_info.components["libthrift"].builddirs = [os.path.join("lib", "cmake")]
```
Comment (SpaceIm) [inline recipes/thrift/all/conanfile.py:195]: No, it's useless and actually a regression.

File: recipes/thrift/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/thrift/all/conanfile.py:146]: No, files created in lib/cmake for this recipe are internal details of cmake_find_package

**Pull request merged**